First draft of docs for trying OSCAR online (#788)
oscar-system/oscar-website: try-online/index.md

---
layout: page
permalink: /try-online/
---

# Try OSCAR Online

> OSCAR can be tried online without installation, on the NFDI jupyter hub.

Reviewer: ```suggestion
OSCAR can be tried online without installation, on the NFDI Jupyter hub.
```

The NFDI Jupyter hub allows 130 credits worth of usage per user per day. Trying out OSCAR requires
40 credits per hour. So, OSCAR can be tried out online free of cost for slightly more than 3 hours
in a day.

Use the following steps:

1. Go to
   <https://hub.nfdi-jupyter.de/workshops/oscar-latest>,

>    and login. Select "Helmholtz AAI" if no other choice applies to you.

Reviewer: ```suggestion
   and log in. Select "Helmholtz AAI" if no other choice applies to you.
```

   ![Main Login](helmholtz-login.png){: width="50%" }
   ![AAI Choice](choose-helmholtz.png){: width="50%" }

2. Select your academic institution, or Google/Github/ORCID to identify yourself. Follow the
   authentication workflow of your chosen service (different for each service). It should be safe to
   allow permissions asked for along this workflow.

   If this is the first time you are logging in to Helmholtz AAI, there may be additional sign up
   screens. If at any point you get stuck, try starting over by visiting (not using the back button)
   <https://hub.nfdi-jupyter.de/workshops/oscar-latest>.
   ![](pick-github.png){: width="50%" }

4. You should now see a list similar to the one depicted in the screenshow
   below, with a row saying "Try out the latest OSCAR in the cloud" and a
   "Start" button to the right of it. Click on that "Start" button to start
   OSCAR online. This may take some time, up to 10 minutes. You will be

>    automatically redirected to a jupyter interface when it is ready

Reviewer: Maybe do a global search and replace to make sure "Jupyter" is spelled consistently.

---
layout: page
permalink: /try-online/
---

# Try OSCAR Online

OSCAR can be tried online without installing, on the NFDI jupyter hub. We provide a docker image
with a precompiled ahead of time version of OSCAR. This strategy side steps lenghty online 

> precompiling, and allows an online "try it before you buy it" version of OSCAR.

Reviewer 2: ```suggestion
OSCAR can be tried online without installation, on the NFDI jupyter hub.
```
This is wayy to technical for new users

The NFDI Jupyter hub allows 130 credits worth of usage per user per day. Trying out OSCAR requires
40 credits per hour. So, OSCAR can be tried out online free of cost for slightly more than 3 hours
in a day.

Use the following steps:

1. Go to
   [https://hub.nfdi-jupyter.de/workshops/oscar-latest](https://hub.nfdi-jupyter.de/workshops/oscar-latest)

>    , and login. Select "Helmholtz AAI" if no other choice applies to you.

Reviewer: The comma must not be separated from the link by spaces, otherwise the spaces are rendered in the result. Also, use `<LINK>` instead of `[LINK](LINK)`
```suggestion
   <https://hub.nfdi-jupyter.de/workshops/oscar-latest>,
   and login. Select "Helmholtz AAI" if no other choice applies to you.
```

   ![Main Login](helmholtz-login.png){: width="50%" }
   ![AAI Choice](choose-helmholtz.png){: width="50%" }

2. Select your academic institution, or Google/Github/ORCID to identify yourself. Follow the
   authentication workflow of your chosen service (different for each service). It should be safe to
   allow permissions asked for along this workflow.

   If this is the first time you are logging in to Helmholtz AAI, there may be additional sign up
   screens. If at any point you get stuck, try starting over by visiting (not using the back button)

>    [https://hub.nfdi-jupyter.de/workshops/oscar-latest](https://hub.nfdi-jupyter.de/workshops/oscar-latest).

Reviewer: ```suggestion
   <https://hub.nfdi-jupyter.de/workshops/oscar-latest>.
```

   As the Helmholtz specific setup needs to be done only once, this should not get you stuck into an

>    infinite loop!

Reviewer: If we can't make a definite promise, we shouldn't mention it: "using this tool might not kill you!" is not great PR 😛 
```suggestion
```

   ![](pick-github.png){: width="50%" }

4. Start the workshop image. This may take some time, up to 10 minutes (depending on how long it
   takes to download the OSCAR docker image). You will be automatically redirected to a jupyter

>    interface when it is ready.

Reviewer 2: ```suggestion
4. Start the workshop image. This may take some time, up to 10 minutes.
    You will be automatically redirected to a jupyter interface when it is ready.
```
too much technical details

Reviewer: I agree but even the revised version is too technical (no offense): normal people have one definition for "image" only, and that's the graphical/visual one, not a "docker image". Actually, *I* do not understand these instruction, as there are no images (recognizable as such, including "Docker images") on the screen I arrived at.

How about
```suggestion
4. You should now see a list similar to the one depicted in the screenshow
   below, with a row saying "Try out the latest OSCAR in the cloud" and a
   "Start" button to the right of it. Click on that "Start" button to start
   OSCAR online. This may take some time, up to 10 minutes. You will be
   automatically redirected to a jupyter interface when it is ready
```

Reviewer: Also, for me it was more like 30 seconds, while the "10 minutes sounds very scary. Perhaps we should qualify this: "May take up to 10 minutes but usually is much faster" ?

   ![](hit-start.png){: width="50%" }

> 5. Select the option in the notebook section which says "Oscar".

Reviewer: what is the difference between the OSCAR variants in the notebook versus the console section?  In any case, perhaps we should add a little text after the image which says: "If you accidentally started the OSCAR entry in the "console" section (instead of the "notebook" section), or any other of the listed entries, don't worry: you can just do INSERT INSTRUCTION HERE.

>    ![Oscar selection in notebook section](oscar-notebook.png){: width="50%" }

Reviewer 2: Why is the kernel named "Oscar 1.12" even though it does not provide version 1.12? Wouldn't it be better to name it "OSCAR with julia 1.12"?

> 6. Try to execute a simple statement to check that Julia kernel has started and connected to the

Reviewer: ```suggestion
6. Try to execute a simple statement to check that the Julia kernel has started and is connected to the
```

>    notebook. `println(4)`, for example. (This may take multiple minutes as the server finishes

Reviewer 2: ```suggestion
   notebook, `println(4)`, for example. (This may take multiple minutes as the server finishes
```

---
layout: page
permalink: /try-online/
---

# Try OSCAR Online

OSCAR can be tried online without installation, on the NFDI Jupyter hub.

The NFDI Jupyter hub allows 130 credits worth of usage per user per day. Trying out OSCAR requires
40 credits per hour. So, OSCAR can be tried out online free of cost for slightly more than 3 hours
in a day.

Use the following steps:

1. Go to
   <https://hub.nfdi-jupyter.de/workshops/oscar-latest>,

>    and log in. Select "Helmholtz AAI" if no other choice applies to you.

Reviewer: the website says "Sign In", and you do show an image of that. Still, how about
```suggestion
   and log in using the "Sign In" button. On the page that appears
   next, select "Helmholtz AAI" if no other choice applies to you.
```

---
layout: page
permalink: /try-online/
---

# Try OSCAR Online

OSCAR can be tried online without installation, on the NFDI Jupyter hub.

The NFDI Jupyter hub allows 130 credits worth of usage per user per day. Trying out OSCAR requires
40 credits per hour. So, OSCAR can be tried out online free of cost for slightly more than 3 hours
in a day.

Use the following steps:

1. Go to
   <https://hub.nfdi-jupyter.de/workshops/oscar-latest>,
   and log in using the "Sign In" button. On the page that appears

>    next, select "Helmholtz AAI" if no other choice applies to you.

Reviewer: Actually the next screen has a list but not "Helmholtz AAI" in it, but there is "Gast-Zugang / Guest access" (this seems to be new? There is also an option "Mardi" but this leads me to a login screen for which I have no account/password)

Reviewer: My comment above still applies

Reviewer 3: Acknowledged; am working on it. (After talking to @aaruni, we agreed that I will take this over and finalize this PR.)

   ![Main Login](helmholtz-login.png){: width="50%" }
   ![AAI Choice](choose-helmholtz.png){: width="50%" }

2. Select your academic institution, or Google/Github/ORCID to identify yourself. Follow the
   authentication workflow of your chosen service (different for each service). It should be safe to
   allow permissions asked for along this workflow.

>    If this is the first time you are logging in to Helmholtz AAI, there may be additional sign up

Reviewer: So this time I used "Guest Access" and then "RPTU" not Helmoltz AAI. That did seem to work

Reviewer 3: I second this.

   screens. If at any point you get stuck, try starting over by visiting (not using the back button)
   <https://hub.nfdi-jupyter.de/workshops/oscar-latest>.
   ![](pick-github.png){: width="50%" }

4. You should now see a list similar to the one depicted in the screenshow
   below, with a row saying "Try out the latest OSCAR in the cloud" and a

>    "Start" button to the right of it. Click on that "Start" button to start

Reviewer: As a matter of fact, that's not what happened for me -- or rather, it did, but there was an intermediate step where I saw this:

<img width="1491" height="254" alt="Image" src="https://github.com/user-attachments/assets/3a62f626-e42d-4f42-95b1-222cfe86ca93" />

I has to click on "Use" there to see the page with "Start" you are describing above.

Reviewer: After "Start" (it's still going on as I wrote), I also see the message "JSC-Cloud ( Credits: 250 / 250 per day )"  -- so not 130 credits as you say above?! But perhaps that's because I chose "RPTU" instead of Helmholtz AAI?

Reviewer 3: I just tried it out myself, and could not see the above message that you stated.

Also, I now see 500 credits. (Logged in as Guest and then RPTU.)

Presumably, the interface changes/credits change a lot and my depend on the user/OS? I think we should thus keep the instructions as concrete as possible but also sufficiently abstract so we do not have to update the instructions with every GUI change. I will think about it.

   OSCAR online. This may take some time, up to 10 minutes. You will be
   automatically redirected to a Jupyter interface when it is ready
   ![](hit-start.png){: width="50%" }

5. Select the option in the notebook section which says "Oscar".
   ![Oscar selection in notebook section](oscar-notebook.png){: width="50%" }

>    If you accidentally started the OSCAR entry in the "console" section (instead of the "notebook"

Reviewer: Hmm, OSCAR versus Oscar...

---
layout: page

> permalink: /try-online/

Author: ```suggestion
permalink: /try-online/
title: Try OSCAR Online
```

I am not sure how its currently inferring a title for the page, but IMO, best to make it explicit.

---

> # Try OSCAR Online

Author: ```suggestion

```

@HereAround the title appears twice because it is rendered twice: once because of the page title, and once again, because of this line in the md file.

Removing this line should fix it.
